### PR TITLE
Fix bug when reading system.parts using UUID (issue 61220).

### LINKED ISSUE #61220: Unable to select anything in `system.parts` using UUID
When trying to select a table name or a part name from `system.parts` table using `uuid` we don't get any output, starting from ClickHouse 24.

For example,

```sql
CREATE TABLE users (uid Int16, name String, age Int16) ENGINE=MergeTree ORDER BY uid PARTITION BY uid;

INSERT INTO users VALUES (1231, 'John', 33);
INSERT INTO users VALUES (6666, 'Ksenia', 48);

SELECT uuid, name from system.parts WHERE table = 'users';

SELECT table from system.parts WHERE uuid = '00000000-0000-0000-0000-000000000000';
``` 

This query in 23.12 would output expected

```sql
00000000-0000-0000-0000-000000000000	1231_1_1_0
00000000-0000-0000-0000-000000000000	6666_2_2_0
users
users
```

But in >=24.1 we only get the output from the 

```sql
SELECT uuid, name from system.parts WHERE table = 'users';

00000000-0000-0000-0000-000000000000	1231_1_1_0
00000000-0000-0000-0000-000000000000	6666_2_2_0
```

and the following query just returns an empty value, when it should return the name of the table which the part belongs to and has the given uuid:

```sql
SELECT table from system.parts WHERE uuid = '00000000-0000-0000-0000-000000000000';
```

Here's the example for 24.2: https://fiddle.clickhouse.com/e779eee6-48fc-4cac-b16a-9d58e0700d24
Here's how it was behaving before the 24 release: https://fiddle.clickhouse.com/c07dc612-876f-4b15-bef7-8ee63ae68ee3

### PR DESCRIPTION
**Pull request by wudanzy**
During the predicates pushing down of system.parts table, there is a
logic to use that predicates to filter on the storages. This makes sense
if the predicate is on the `database`, `table`, `engine`. But it can be
problem if the predicate is about `uuid`, because storage UUID is
apparently different from parts UUID.

Rename the column name from `uuid` to `storage_uuid` fixed this.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix incorrect results when filtering `system.parts` or `system.parts_columns` using UUID.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit adaf262dc09ff08628d54d5ffd64f7ea5053482d with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61479/adaf262dc09ff08628d54d5ffd64f7ea5053482d/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: tests/queries/0_stateless/03010_read_system_parts_table_test.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+DROP TABLE IF EXISTS users;
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/03010_read_system_parts_table_test.sql:1]: Could you please also add tests for `parts_columns`, `projection_parts` and `projection_parts_columns`? They all use the same base class, in theory they should behave the same, but better to have a test for them.

**Review by antaljanosbenjamin — COMMENTED** [review]
The changes looks good, but could you please merge/rebase master? I think the clang-tidy failures are fixed on master.

File: tests/queries/0_stateless/03010_read_system_parts_table_test.sql (hunk 1 above)
Comment (wudanzy) [reply tests/queries/0_stateless/03010_read_system_parts_table_test.sql:1]: Hi János, thanks for the suggestion! Just had a try, but found out that [`projection_parts`](https://github.com/ClickHouse/ClickHouse/blob/e82cf5d5ba62ec4e13288052e690228707e9de4e/src/Storages/System/StorageSystemProjectionParts.cpp#L21) and [`projection_parts_columns`](https://github.com/ClickHouse/ClickHouse/blob/e82cf5d5ba62ec4e13288052e690228707e9de4e/src/Storages/System/StorageSystemProjectionPartsColumns.cpp#L21) do not have a column called `uuid`. 

Added test for `projection_parts`.

**Comment by wudanzy** [comment]
Hi János, I am new to Clickhouse, could you please share a guidance on how to debug failing tests?

For Bugfix Validation that checks that either a new test (functional or integration) or there some changed tests that fail with the binary built on master branch, I am sure it did fail. https://fiddle.clickhouse.com/6891df0b-0a5d-42c7-a104-4a58d93799c1

**Comment by antaljanosbenjamin** [comment]
Hi Dan!

For the bugfix validation the issue is not in your code. [This PR](https://github.com/ClickHouse/ClickHouse/pull/61363) will fix it. Let's wait a bit, hopefully it will get merged soon. Then you can merge a master to your branch so the updated config should run the bugfix check properly.

The RISC-V build and the single stateless test failure is also not related to your code.

**Comment by antaljanosbenjamin** [comment]
Everything looks fine:
- build check failed on RISC-V which is a known issue
- bugfix failed because of a known issue, tested locally the the test only passes if I remove the filters on `uuid`, so the fix probably right
- the failed stateless test (`02481_async_insert_race_long`) is flaky

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**